packages/headless/src/controllers/result-list/headless-result-link-analytics.ts

```diff
@@ -0,0 +1,77 @@
+import {Result} from '../../api/search/search/result';
+import {Engine} from '../../app/headless-engine';
+import {logDocumentOpen} from '../../features/result/result-analytics-actions';
+
+// 1 second is a reasonable amount of time to catch most longpress actions
+const defaultDelay = 1000;
+
+type ResultLinkAnalyticsOptions = {
+  /**
+   * The result for which to log analytics.
+   */
+  result: Result;
+  /**
+   * The amount of time to wait before selecting the result after calling `beginDelayedSelect`.
+   */
+  delay?: number;
+};
+
+type ResultLinkAnalyticsProps = {
+  /** The options for the `ResultLinkAnalytics` controller. */
+  options: ResultLinkAnalyticsOptions;
+};
+
+/**
+ * The `ResultLinkAnalytics` headless controller helps more easily send analytics when a search result is selected.
+ */
+export type ResultLinkAnalytics = ReturnType<typeof buildResultLinkAnalytics>;
+
+export function buildResultLinkAnalytics(
+  engine: Engine,
+  props: ResultLinkAnalyticsProps
+) {
+  let wasOpened = false;
```

Comment:
What should we do if a link is opened in a new tab, closed, and opened again? Is it correct to log only one analytics event?

Reply:
I don't think there's a point in knowing whether a link was opened once or more than once. I don't think it's necessarily bad either though.

Reply:
Maybe we could simplify by removing the `wasOpened` flag then?
From an ML perspective, I would imagine if for a given query a result is clicked two times, this could be valuable info?

Reply:
This can screw up dashboards though. 

This would incorrectly report a higher average "click per query". 
One result opened 1 time or opened 50 times should be reported only once.

So it does have an importance.

And even there we are not perfect, because if somebody opens a link, navigate away from the search page (so: headless gets unloaded from memory/re-initialized), comes back, re-execute the same query, and re-open it, it should in theory still be reported as one click.

But that's kind of corner case... Anyway 🤷